### ISSUE TITLE
Error if  `DEFAULT_CHEAT_DIR` contains `~` or envvars

### ISSUE THREAD

**Issue by brutus** [chrisallenlane/cheat#310]
cheat bails with _"Could not create DEFAULT_CHEAT_DIR"_ if the `DEFAULT_CHEAT_DIR` contains environment variables or `~`.

See https://github.com/chrisallenlane/cheat/pull/309 for a fix.

**Comment by lucaswerkmeister** [comment]
Why should `DEFAULT_CHEAT_DIR` contain unexpanded environment variables or tildes? Those should be expanded by your shell when you set the variable, not by the program that interprets the variable. For example, Bash’s `HISTFILE` or less’ `LESSHISTFILE` also don’t support unexpanded tildes.

**Comment by brutus** [comment]
Good catch: My situation was that I use a file containing all my environment variables, which is sourced by the bash profile and distributed to different hosts with Ansible. Turns out, I quoted the string for `DEFAULT_CHEAT_DIR` so `~` wasn't expanded. Cheat was just the first program that ever complained… I seems I will never remember when and what to quote in bash :confused:

The following seems to work for me:

```bash
if [[ ~/.local/share/cheat/ ]]; then
  export CHEATCOLORS=1
  export DEFAULT_CHEAT_DIR=~/.local/share/cheat
fi
```

**Comment by chrisallenlane** [comment]
Closing as resolved by #309.

**Comment by lucaswerkmeister** [comment]
Well, I still think this *shouldn’t* ~~be~~ have been fixed, but whatever.
